auction-server/src/opportunity/repository/mod.rs

```diff
@@ -0,0 +1,94 @@
+use {
+    super::entities,
+    crate::kernel::entities::PermissionKey,
+    ethers::types::Address,
+    std::{
+        collections::HashMap,
+        ops::Deref,
+    },
+    tokio::sync::RwLock,
+};
+
+mod add_opportunity;
+mod add_spoof_info;
+mod get_opportunities;
+mod get_opportunities_by_permission_key;
+mod get_opportunities_by_permission_key_and_id;
+mod get_spoof_info;
+mod models;
+mod opportunity_exists;
+mod remove_opportunity;
+
+#[derive(Debug)]
+pub struct Repository<T: Cache> {
+    pub cache: T,
+}
+
+pub trait Cache: Deref<Target = CacheCoreFields<Self::Opportunity>> {
```

Comment:
Maybe rename to `InMemoryStore`? Cache misrepresents all the logic we have put inside this